plugin/cmd/kube-scheduler/app/options/options_test.go

```diff
@@ -0,0 +1,103 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package options
+
+import (
+	"reflect"
+	"testing"
+	"time"
+
+	"github.com/spf13/pflag"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/diff"
+	"k8s.io/kubernetes/pkg/apis/componentconfig"
+)
+
+func TestAddFlags(t *testing.T) {
+	f := pflag.NewFlagSet("addflagstest", pflag.ContinueOnError)
+	s := NewSchedulerServer()
+	s.AddFlags(f)
+
+	args := []string{
+		"--address=192.168.4.20",
+		"--algorithm-provider=FooProvider",
+		"--contention-profiling=true",
+		"--failure-domains=kubernetes.io/hostname",
+		"--hard-pod-affinity-symmetric-weight=0",
+		"--kube-api-burst=80",
+		"--kube-api-content-type=application/vnd.kubernetes.protobuf",
+		"--kube-api-qps=40.0",
+		"--kubeconfig=/foo/bar",
+		"--leader-elect=true",
+		"--leader-elect-lease-duration=20s",
+		"--leader-elect-renew-deadline=15s",
+		"--leader-elect-resource-lock=endpoints",
+		"--leader-elect-retry-period=3s",
+		"--lock-object-name=foo",
```

Comment:
Could you use something other than "foo" and "bar" repeatedly? The reason is that we want to make sure that one config is not taken for another one by mistake. For example, we want to ensure that "kubeconfig" file is not taken as "policy-config-file". The current values of the fields does not verify this.
You may want to use something like "test-lock-obj" here.

Reply:
Sure, I'll do that.